### PR TITLE
refactor(FlowCellDirectoryData)

### PR DESCRIPTION
**Pull request by ChrOertlin**
### Added

- Parent class `InstrumentRunDirectoryData` empty for now. Parent class to collect similarities for other run instruments.

### Changed

- Renames `FlowCellDirectoryData` to `IlluminaRunDirectoryData`. Reason with the implementation of new instruments we will get more different `FlowCellDirectoryData` for example `Pacbio` and `Nanopore` that are inherently different than `Illumina`'s.

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by diitaz93** [comment]
What happened with `FlowCell`? 👀

**Comment by ChrOertlin** [comment]
> What happened with `FlowCell`? 👀

Technically it is not a flow cell directory but the output of a sequencing run. The flow cell is just a cartridge that we use in the machine. 

This here is more of a proposal that we can discuss.

**Comment by henrikstranneheim** [comment]
I think this sounds like a good idea to decouple the naming from what it was processed on to what it represents in a larger scope.

File: cg/models/instrument_run_directory_data/instrument_run_directory_data.py [hunk 1]
```diff
@@ -36,7 +36,11 @@
 }
 
 
-class FlowCellDirectoryData:
+class InstrumentRunDirectoryData:
+    pass
+
+
+class IlluminaRunDirectoryData(InstrumentRunDirectoryData):
```
Comment (diitaz93) [inline cg/models/instrument_run_directory_data/instrument_run_directory_data.py:43]: As agreed on the standup
```suggestion
class IlluminaRunDirectory(InstrumentRunDirectoryData):
```

File: cg/cli/backup.py [hunk 2]
```diff
@@ -27,8 +27,8 @@
 )
 from cg.meta.tar.tar import TarAPI
 from cg.models.cg_config import CGConfig
-from cg.models.flow_cell.flow_cell import (
-    FlowCellDirectoryData,
+from cg.models.illumina_run_directory_data.illumina_run_directory import (
```
[the remark this answers is not in the document]
Comment (diitaz93) [reply cg/cli/backup.py:30]: Other option could be `device_directories`

File: cg/cli/demultiplex/demux.py [hunk 3]
```diff
@@ -24,54 +24,56 @@
...
+from cg.models.illumina_run_directory_data.illumina_run_directory import (
+    IlluminaRunDirectory,
+)
 
 LOG = logging.getLogger(__name__)
 
 
 @click.command(name="all")
-@click.option("--flow-cells-directory", type=click.Path(exists=True, file_okay=False))
+@click.option("--sequencing-runs-directory", type=click.Path(exists=True, file_okay=False))
```
[the remark this answers is not in the document]
Comment (diitaz93) [reply cg/cli/demultiplex/demux.py:35]: I have not found the place where this path is defined though, maybe Im wrong, but better double check

Comment (ChrOertlin) [inline cg/cli/demultiplex/demux.py:35]: This one is not specified in the systemd command so this change should be safe.

File: tests/meta/backup/test_meta_backup.py [hunk 4]
```diff
@@ -692,12 +694,12 @@ def test_create_copy_complete_file_exist(
...
 def test_create_copy_complete_file_does_not_exist(
     backup_api: BackupAPI,
-    novaseq_x_flow_cell: FlowCellDirectoryData,
+    novaseq_x_flow_cell: IlluminaRunDirectory,
 ):
     """Tests creating a copy complete file in the flow cell directory. There are two cases: when
     the file exists and when it does not exist."""
 
-    # GIVEN a flow cell that has been decrypted in flow_cell directory
+    # GIVEN a flow cell that has been decrypted in illumina_run_directory_data directory
```
Comment (diitaz93) [inline tests/meta/backup/test_meta_backup.py:702]:
```suggestion
    # GIVEN a flow cell that has been decrypted in illumina_run_directory directory
```

File: cg/cli/backup.py
Comment (ChrOertlin) [inline cg/cli/backup.py:1]: Not touching any of the CLI signatures here. 
This part needs refactoring in the next days with the use of the new instrument runs table to remove the use of the current `FlowCell` will revisit this then in more detail and smaller scope.

File: cg/cli/backup.py [hunk 5]
```diff
@@ -206,7 +210,9 @@ def archive_spring_file(config: CGConfig, spring_file_path: str, dry_run: bool):
 @DRY_RUN
 @click.option("-s", "--sample-id", "object_type", flag_value="sample", type=str)
 @click.option("-c", "--case-id", "object_type", flag_value="case", type=str)
-@click.option("-f", "--flow-cell-id", "object_type", flag_value="flow_cell", type=str)
+@click.option(
+    "-f", "--flow-cell-id", "object_type", flag_value="illumina_run_directory_data", type=str
```
Comment (diitaz93) [inline cg/cli/backup.py:214]: I would say limit this PR only to the renaming of the model, and do CLI command changes in another PR

File: cg/cli/demultiplex/demux.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ChrOertlin) [reply cg/cli/demultiplex/demux.py:35]: In servers the cli command is called as:

`cg demultiplex finish flow-cell all` There is no path given. To be quite honest. this function should not require a passable directory, since the directory is pulled from the config.

File: cg/models/instrument_run_directory_data/instrument_run_directory_data.py (hunk 1 above)
Comment (diitaz93) [reply cg/models/instrument_run_directory_data/instrument_run_directory_data.py]: I see it was changed, resolving

File: cg/models/instrument_run_directory_data/instrument_run_directory_data.py [hunk 6]
```diff
@@ -36,7 +36,11 @@
 }
 
 
-class FlowCellDirectoryData:
+class InstrumentRunDirectoryData:
```
[the remark this answers is not in the document]
Comment (diitaz93) [reply cg/models/instrument_run_directory_data/instrument_run_directory_data.py]: I see this was implemented, resolving

File: cg/cli/demultiplex/demux.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (diitaz93) [reply cg/cli/demultiplex/demux.py:35]: I would say leave this change for another PR with only changes to the CLI commands

File: cg/meta/demultiplex/demux_post_processing.py
Comment (ChrOertlin) [inline cg/meta/demultiplex/demux_post_processing.py:1]: skipping this as it will be deprecated soon

**Comment by ChrOertlin** [comment]
will be split into multiple smaller but used as reference.
